pkg/server/api/entry/v1/service_test.go

```diff
@@ -2,29 +2,246 @@ package entry_test
 
 import (
 	"context"
+	"errors"
 	"testing"
+	"time"
 
+	"github.com/sirupsen/logrus"
 	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+	"github.com/spiffe/spire/pkg/server/api"
 	"github.com/spiffe/spire/pkg/server/api/entry/v1"
 	"github.com/spiffe/spire/pkg/server/api/rpccontext"
 	entrypb "github.com/spiffe/spire/proto/spire-next/api/server/entry/v1"
+	"github.com/spiffe/spire/proto/spire-next/types"
+	"github.com/spiffe/spire/proto/spire/common"
+	"github.com/spiffe/spire/proto/spire/server/datastore"
 	"github.com/spiffe/spire/test/fakes/fakedatastore"
 	"github.com/spiffe/spire/test/spiretest"
+	"github.com/stretchr/testify/require"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
 )
 
-type serviceTest struct { //nolint: unused,deadcode
+var (
+	ctx         = context.Background()
+	td          = spiffeid.RequireTrustDomainFromString("example.org")
+	federatedTd = spiffeid.RequireTrustDomainFromString("domain1.org")
+)
+
+func TestGetEntry(t *testing.T) {
+	test := setupServiceTest(t)
+	defer test.Cleanup()
+
+	test.createBundles(t)
+
+	parent := td.NewID("foo")
+	entry1SpiffeID := td.NewID("bar")
+	expiresAt := time.Now().Unix()
+	entry1, err := test.ds.CreateRegistrationEntry(ctx, &datastore.CreateRegistrationEntryRequest{
```

Comment:
nit: seems a tiny bit confusing to have `entry1` without an `entry2`
```suggestion
	goodEntry, err := test.ds.CreateRegistrationEntry(ctx, &datastore.CreateRegistrationEntryRequest{
```